src/main/java/wanya/parser/Parser.java

```diff
@@ -0,0 +1,109 @@
+package wanya.parser;
+
+import wanya.task.TaskList;
+import wanya.Ui;
+import wanya.WanyaException;
+
+import java.time.DateTimeException;
+
+/**
+ * Represents a parser that helps to understand user's input commands.
+ */
+public class Parser {
+    /**
+     * Checks for empty description of tasks inputted.
+     *
+     * @param inputs list of Strings that user has inputted.
+     * @param command the first word of the user input.
+     * @throws WanyaException if description of task is empty.
+     */
+    public static void checkTask(String[] inputs, String command) throws WanyaException {
+        //handle the error where no task name
+        if (inputs.length == 1 || inputs[1].trim().startsWith("/at")
+                || inputs[1].trim().startsWith("/by") || inputs[1].trim().equals("")) {
+            throw new WanyaException("The description of " + command + " cannot be empty");
+        }
+    }
+
+    /**
+     * Checks for task number to be valid and within the size of task list.
+     *
+     * @param inputs list of Strings that user has inputted.
+     * @param tasks TaskList that contains all the tasks.
+     * @return task number if it is valid.
+     * @throws WanyaException if task number provided is negative or
+     *         greater than the number of tasks in TaskList.
+     */
+    public static int checkTaskNumber(String[] inputs, TaskList tasks) throws WanyaException {
+        //handle error without task number
+        if (inputs.length == 1 || inputs[1].trim().equals("")) {
```

Comment:
should input[1] be extracted out and be referenced as a variable instead? Since you used it a few times after.